client/components/UserAutocomplete/UserAutocomplete.tsx

```diff
@@ -28,26 +28,31 @@ type Props = OwnProps & typeof defaultProps;
 class UserAutocomplete extends Component<Props, State> {
 	static defaultProps = defaultProps;
 
+	runUserQuery = (queryValue) => {
+		apiFetch(`/api/search/users?q=${queryValue}`).then((result) => {
+			const { usedUserIds } = this.props;
+			this.setState({
+				items: result.filter((item) => !usedUserIds.includes(item.id)),
+			});
+		});
+	};
+
 	constructor(props: Props) {
 		super(props);
 		this.state = {
 			items: [],
 			queryValue: '',
 		};
+		this.runUserQuery = debounce(this.runUserQuery, 300);
 		// @ts-expect-error ts-migrate(2339) FIXME: Property 'inputRef' does not exist on type 'UserAu... Remove this comment to see the full error message
 		this.inputRef = undefined;
 		this.handleSelect = this.handleSelect.bind(this);
 	}
 
 	componentDidUpdate(_: Props, prevState: State) {
 		const { queryValue } = this.state;
-		if (queryValue !== prevState.queryValue) {
-			apiFetch(`/api/search/users?q=${queryValue}`).then((result) => {
-				const { usedUserIds } = this.props;
-				this.setState({
-					items: result.filter((item) => !usedUserIds.includes(item.id)),
-				});
-			});
+		if (queryValue !== prevState.queryValue && queryValue !== '') {
+			this.runUserQuery(queryValue);
```

Comment:
It felt like grabbing the query value and comparing it with the previous state still needed to happen in componentDidUpdate? But it was just a feeling. Can move into `runUserQuery` if it's cleaner.